Quorum/checks/new_listing.py

```diff
@@ -26,168 +29,18 @@ def new_listing_check(self) -> None:
         no new listings were found.
         """
         functions = self._get_functions_from_source_codes()
-
-        if "newListings" in functions or "newListingsCustom" in functions:
-            self._handle_new_listings(functions)
+        if function := functions.get("newListings", functions.get("newListingsCustom")):
+            pp.pretty_print(f"New listings detected for {self.proposal_address}", pp.Colors.WARNING)
+            self._write_to_file(self.check_path, data=function)
         else:
             pp.pretty_print(f"No new listings detected for {self.proposal_address}", pp.Colors.INFO)
-
+    
     def _get_functions_from_source_codes(self) -> dict:
         """
         Retrieves functions from the source codes.
-        This method iterates over the source codes and collects all functions defined in them.
-        Returns:
-            dict: A dictionary where keys are function names and values are function nodes.
+        This method retrieves functions from the source codes and returns them in a dictionary.
```

Comment:
If they have multiple contracts where on each they have the same function than yes. but first i am not sure its possible. and second we don't care because we only want to check the existence of some functions in the contract.

Reply:
Listing can be done only once right?
Or is there any chance of a big proposal that suggest 2 new listings? 🤔

Reply:
Only once. its should override specific function on the specific contract.